front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -154,10 +160,26 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
     [projectedTrainsById]
   );
 
-  const timetableItemUsedForProjection = useMemo(
-    () => (trainIdUsedForProjection ? timetableItemsById.get(trainIdUsedForProjection) : undefined),
-    [trainIdUsedForProjection, timetableItems]
-  );
+  const pathUsedForProjection = useMemo(() => {
+    if (!trainIdUsedForProjection) return undefined;
+    if (!isOccurrenceId(trainIdUsedForProjection)) {
+      return timetableItemsById.get(trainIdUsedForProjection)?.path;
+    }
+
+    const pacedTrainId = extractPacedTrainIdFromOccurrenceId(trainIdUsedForProjection);
+    const pacedTrain = timetableItemsById.get(pacedTrainId);
+    if (!pacedTrain || !isPacedTrainResponseWithPacedTrainId(pacedTrain)) throw new Error();
+
+    let exception: PacedTrainException | undefined;
+    if (isIndexedOccurrenceId(trainIdUsedForProjection)) {
+      const index = extractOccurrenceIndexFromOccurrenceId(trainIdUsedForProjection);
+      exception = pacedTrain.exceptions.find((e) => e.occurrence_index === index);
+    } else {
+      const key = extractExceptionIdFromOccurrenceId(trainIdUsedForProjection);
+      exception = pacedTrain.exceptions.find((e) => e.key === key);
+    }
```

Comment:
I think you could simplify this block by extracting the identifier (index or key) using a ternary expression, and then using a single find() call with a conditional property check.
This would avoid the duplication and make the logic a bit easier to follow.